ethereum/p2p/src/main/java/tech/pegasys/pantheon/ethereum/p2p/network/DefaultP2PNetwork.java

```diff
@@ -242,14 +257,22 @@
         "netty_boss_pending_tasks",
         "The number of pending tasks in the Netty boss event loop",
         pendingTaskCounter(boss));
+  }
 
-    subscribeDisconnect(peerDiscoveryAgent);
-    subscribeDisconnect(peerBlacklist);
-    subscribeDisconnect(connections);
+  public static Builder builder() {
+    return new Builder();
+  }
 
-    this.keyPair = keyPair;
-    this.subProtocols = config.getSupportedProtocols();
+  private Supplier<Integer> pendingTaskCounter(final EventLoopGroup eventLoopGroup) {
+    return () ->
+        StreamSupport.stream(eventLoopGroup.spliterator(), false)
+            .filter(eventExecutor -> eventExecutor instanceof SingleThreadEventExecutor)
+            .mapToInt(eventExecutor -> ((SingleThreadEventExecutor) eventExecutor).pendingTasks())
+            .sum();
+  }
 
+  /** Start listening for incoming connections */
+  private void startListening() {
```

Comment:
This diff has ended up looking pretty confusing.  I'm just moving this listening logic out of the constructor and into a separate method.  I also did some reordering of the constructor logic so that it reads better.